rpc/filter_system.go

```diff
@@ -0,0 +1,433 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	tmquery "github.com/tendermint/tendermint/libs/pubsub/query"
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+	coretypes "github.com/tendermint/tendermint/rpc/core/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/eth/filters"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	evmtypes "github.com/cosmos/ethermint/x/evm/types"
+)
+
+var (
+	txEvents     = tmtypes.QueryForEvent(tmtypes.EventTx).String()
+	evmEvents    = tmquery.MustParse(fmt.Sprintf("%s='%s' AND %s.%s='%s'", tmtypes.EventTypeKey, tmtypes.EventTx, sdk.EventTypeMessage, sdk.AttributeKeyModule, evmtypes.ModuleName)).String()
+	headerEvents = tmtypes.QueryForEvent(tmtypes.EventNewBlockHeader).String()
+)
+
+// EventSystem creates subscriptions, processes events and broadcasts them to the
+// subscription which match the subscription criteria using the Tendermint's RPC client.
+type EventSystem struct {
+	ctx    context.Context
+	client rpcclient.Client
+
+	// light client mode
+	lightMode bool
+
+	index filterIndex
+
+	// Subscriptions
+	txsSub  *Subscription // Subscription for new transaction event
+	logsSub *Subscription // Subscription for new log event
+	// rmLogsSub      *Subscription // Subscription for removed log event
+
+	pendingLogsSub *Subscription // Subscription for pending log event
+	chainSub       *Subscription // Subscription for new chain event
+
+	// Channels
+	install   chan *Subscription // install filter for event notification
+	uninstall chan *Subscription // remove filter for event notification
+
+	// Unidirectional channels to receive Tendermint ResultEvents
+	txsCh         <-chan coretypes.ResultEvent // Channel to receive new pending transactions event
+	logsCh        <-chan coretypes.ResultEvent // Channel to receive new log event
+	pendingLogsCh <-chan coretypes.ResultEvent // Channel to receive new pending log event
+	// rmLogsCh      <-chan coretypes.ResultEvent // Channel to receive removed log event
```

Comment:
is this still needed?

Reply:
yeah, it's used on Geth but idk what is considered a removed log in our case